src/utils/typedData/index.ts

```diff
@@ -102,6 +108,18 @@ const encodeValue = (typedData: TypedData, type: string, data: unknown): [string
     return [type, getStructHash(typedData, type, data as Record<string, unknown>)];
   }
 
+  if (
```

Comment:
Could we separate the logic and if statement?

```
[descriptive variable here] = Object.keys(typedData.types)
      .map((x) => `${x}*`)
      .includes(type)
```

if ([descriptive variable])